base/compiler/abstractinterpretation.jl

```diff
@@ -207,6 +207,40 @@ function abstract_call_method_with_const_args(@nospecialize(f), argtypes::Vector
     return result
 end
 
+# Returns -1 when patype is less complex than catype, 0 when they are equal, 1 when patype is more complex
+# May overapproximate and return 0 (e.g. when the result is indeterminate because the types are not comparable)
```

Comment:
`0` isn't an overapproximation

Reply:
Heh, you caught that, did you ;). You're right of course. That was just a quick way to fix the case I was interested in. `-1` is the correct approximate fallback.